explorer/interpreter/type_checker.cpp

```diff
@@ -2863,6 +2863,24 @@ auto TypeChecker::TypeCheckStmt(Nonnull<Statement*> s,
       CARBON_RETURN_IF_ERROR(TypeCheckStmt(&while_stmt.body(), impl_scope));
       return Success();
     }
+    case StatementKind::For: {
+      auto& for_stmt = cast<For>(*s);
+      ImplScope inner_impl_scope;
+      inner_impl_scope.AddParent(&impl_scope);
+      CARBON_RETURN_IF_ERROR(TypeCheckPattern(&for_stmt.variable_declaration(),
+                                              std::nullopt, inner_impl_scope,
+                                              ValueCategory::Let));
+      CARBON_RETURN_IF_ERROR(
+          TypeCheckExp(&for_stmt.loop_target(), inner_impl_scope));
+
+      CARBON_RETURN_IF_ERROR(
+          ImplicitlyConvert("declaration of for loop variable", impl_scope,
+                            &for_stmt.loop_target(),
+                            &for_stmt.variable_declaration().static_type()));
```

Comment:
Hm, I would have expected that we want to convert the array element type of the type of the loop target, not the type of the loop target itself, here.

Reply:
thanks, solved!